src/esp32/webhook.rs

```diff
@@ -0,0 +1,94 @@
+#[allow(unused_imports)] // need this trait for `get_attribute`
+use crate::common::config::Component;
+use crate::common::config::DynamicComponentConfig;
+use crate::proto::app::v1::ConfigResponse;
+use embedded_svc::{
+    http::{client::Client as HttpClient, Method},
+    io::Write,
+};
+use esp_idf_svc::http::client::{Configuration as HttpConfiguration, EspHttpConnection};
+use log::*;
+use serde_json::json;
+use thiserror::Error;
+
+#[derive(Debug, Error, PartialEq)]
+pub enum WebhookError {
+    #[error("`webhook` not present in board attributes")]
+    NoWebhook,
+    #[error("`webhook-secret` not present in board attributes")]
+    NoWebhookSecret,
+    #[error("Error with webhook request: {0}")]
+    RequestError(String),
+    #[error("Error in config: {0}")]
+    ConfigError(String),
+}
+
+/// Checks the board config for a `webhook` and `webhook-secret` attribute, sending a GET
+/// request with credentials for an SDK to make a connection.
+pub fn handle_webhook(config: ConfigResponse) -> Result<(), WebhookError> {
+    let config = config
```

Comment:
I think this should be handled on the outside and this function should take an instance of `RobotConfig` instead of `ConfigResponse` (I know I've done essentially the same thing in `LocalRobot::new_from_config_response` but I think that was a mistake)